sample_data_config/twitter-ads-api/twitter-ads-api.config.yaml

```diff
@@ -1,7 +1,36 @@
 twitterAdsApi:
-  - dataPipelineId: twitter_ads_api_campaign_elife_journal
+  # - dataPipelineId: twitter_ads_api_campaign_elife_journal
+  #   source:
+  #     resource: '/11/accounts/18ce53up4ko/campaigns'
+  #     secrets:
+  #       parametersFromFile:
+  #         - parameterName: api_key
+  #           filePathEnvName: TWITTER_API_KEY_FILE_PATH
+  #         - parameterName: api_secret
+  #           filePathEnvName: TWITTER_API_SECRET_FILE_PATH
+  #         - parameterName: access_token
+  #           filePathEnvName: TWITTER_ACCESS_TOKEN_FILE_PATH
+  #         - parameterName: access_token_secret
+  #           filePathEnvName: TWITTER_ACCESS_TOKEN_SECRET_FILE_PATH
+  #   target:
+  #     projectName: 'elife-data-pipeline'
+  #     datasetName: '{ENV}'
+  #     tableName: twitter_campaign_details_elife_journal
+
+  - dataPipelineId: twitter_ads_api_campaign_reach_elife_journal
     source:
-      resource: '/11/accounts/18ce53up4ko/campaigns'
+      resource: '/11/stats/accounts/18ce53up4ko/reach/campaigns'
+      paramValueFromBigQuery:
```

Comment:
Just wondering whether this should be moved out of "source" or whether "source" should have a separate api category. Just to make it more clear what affects the API and what not. For example the resource, parameters and secrets are for the API.
Whereas the BigQuery data is separate from the API. It isn't the main source of the pipeline, it rather informs it what to query. A bit like a state file might.
What do you think?

Reply:
structure has been changed as we talked